packages/react-components/src/components/ChatMessageComponent.tsx

```diff
@@ -159,84 +173,136 @@ export const ChatMessageComponent = (props: ChatMessageProps): JSX.Element => {
-        mine={message.mine}
-        timestamp={
-          <Text data-ui-id={ids.messageTimestamp}>
-            {message.createdOn
-              ? showDate
-                ? formatTimestampForChatMessage(message.createdOn, new Date(), strings)
-                : formatTimeForChatMessage(message.createdOn)
-              : undefined}
-          </Text>
-        }
-        details={
-          message.editedOn ? <div className={chatMessageEditedTagStyle(theme)}>{strings.editedTag}</div> : undefined
-        }
-        positionActionMenu={false}
-        actionMenu={!disableEditing && message.status !== 'sending' && message.mine ? actionMenu : undefined}
-      />
-    </div>
+    <>
+      <div ref={messageRef} {...longTouchPressProps}>
+        <Chat.Message
+          className={mergeStyles(messageContainerStyle as IStyle)}
+          styles={messageContainerStyle}
+          content={messageContentItem}
+          author={<Text className={chatMessageDateStyle}>{message.senderDisplayName}</Text>}
+          mine={message.mine}
+          timestamp={
+            <Text data-ui-id={ids.messageTimestamp}>
+              {message.createdOn
+                ? showDate
+                  ? formatTimestampForChatMessage(message.createdOn, new Date(), strings)
+                  : formatTimeForChatMessage(message.createdOn)
+                : undefined}
+            </Text>
+          }
+          details={
+            message.editedOn ? <div className={chatMessageEditedTagStyle(theme)}>{strings.editedTag}</div> : undefined
+          }
+          positionActionMenu={false}
+          actionMenu={actionMenuProps}
+        />
+      </div>
+
+      {chatActionsEnabled && (
+        <ChatMessageActionFlyout
+          hidden={!chatMessageActionFlyoutTarget}
+          target={chatMessageActionFlyoutTarget ?? undefined}
+          onDismiss={() => {
+            setChatMessageActionFlyoutTarget(undefined);
+            setAllowChatActionButtonShow(true);
+          }}
+          onEditClick={() => {
+            setIsEditing(true);
+          }}
+          onRemoveClick={async () => {
+            onDeleteMessage && message.messageId && (await onDeleteMessage(message.messageId));
+          }}
+          strings={strings}
+        />
+      )}
+    </>
   );
 
   return chatMessage;
 };
 
-const MoreMenu = ({
-  onEditClick,
-  onRemoveClick,
-  strings
-}: {
+/**
+ * Props for the Chat.Message action menu.
+ * This is the 3 dots that appear when hovering over one of your own chat messages.
+ */
+const chatMessageActionMenuProps = (menuProps: {
+  menuButtonRef: React.MutableRefObject<HTMLElement | null>;
+  onActionButtonClick: () => void;
+  theme: Theme;
+}): MenuProps => {
+  const menuClass = mergeStyles(chatActionsCSS, {
+    'ul&': { boxShadow: menuProps.theme.effects.elevation4, backgroundColor: menuProps.theme.palette.white }
+  });
+
+  const actionMenuProps: MenuProps = {
+    iconOnly: true,
+    activeIndex: -1,
+    className: menuClass,
```

Comment:
for a11y, you will be mindful of  this commit: https://github.com/Azure/communication-ui-library/pull/1025/commits/44caee79eefbbdf11fe123988975c1afd7961d07

It's already merged to main

Reply:
Thanks! Merged in changes and tested keyboard navigation: 
![image](https://user-images.githubusercontent.com/2684369/140360791-0904f0a4-488b-4ccc-82c3-557bf263b9ff.png)
(note: found a bug where we can't navigate to the reject edits button once edit is selected - filed a bug for it)